test/components/views/dialogs/__snapshots__/MessageEditHistoryDialog-test.tsx.snap

```diff
@@ -77,7 +77,7 @@ exports[`<MessageEditHistory /> should match the snapshot 1`] = `
                   <span
                     class="mx_MessageTimestamp"
                   >
-                    00:00
+                    24:00
```

Comment:
some more odd changes here. 24:00 really shouldn't be a thing.

Reply:
![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/3d1d8199-83a6-421a-896f-f7e0cd700319)

Says who?

Reply:
> we could switch to h11 + h23 hourCycle but then we might end up breaking h12-preferring countries and maybe h24 preferring ones if any exist :/

I don't really follow you. What is "h11 + h23 hourCycle"? (I can't find any documentation for any of these options.) Isn't the UK an h24-preferring country?

It's worth noting that, according to https://en.wikipedia.org/wiki/24-hour_clock#Times_after_24:00, even if 24:00 is acceptable, 24:01 is not.

I don't really feel we can leave this as-is. It feels a problematic regression for users in 12-hour-preferring countries.

Reply:
> I don't really follow you. What is "h11 + h23 hourCycle"? (I can't find any documentation for any of these options.)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat

> The hour cycle to use. Possible values are "h11", "h12", "h23", and "h24". This option can also be set through the hc Unicode extension key; if both are provided, this options property takes precedence.

> Isn't the UK an h24-preferring country?

If it was then 24:59 would be valid and you wouldn't be upset :D

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getHourCycles lacks wide support to query the preferred hour cycles for a given locale. In Chrome it says 23h for `en-GB` and 12h for `en`

![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/e0477ec9-1464-4d1c-bfd9-1547ce224c40)

Which doesn't feel right, I'd have said UK is predominantly 12h.

> I don't really feel we can leave this as-is. It feels a problematic regression for users in 12-hour-preferring countries.

Sorry what's changed for 12h? It seems to only affect 24h users on Chrome who do not expect a 24h hourCycle

Reply:
> > I don't really follow you. What is "h11 + h23 hourCycle"? (I can't find any documentation for any of these options.)
> 
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat
> 
> > The hour cycle to use. Possible values are "h11", "h12", "h23", and "h24". This option can also be set through the hc Unicode extension key; if both are provided, this options property takes precedence.

OK. What's an hour cycle?

> > Isn't the UK an h24-preferring country?
> 
> If it was then 24:59 would be valid and you wouldn't be upset :D

(a) No, 24:59 is never valid; (b) even 24:00 is "weird" imho; (c) I'm perfectly happy with 23:59.
 
> Which doesn't feel right, I'd have said UK is predominantly 12h.

I don't really agree, but it's moot. The point is that there are plenty of countries that prefer 24h in the locale data.

> > I don't really feel we can leave this as-is. It feels a problematic regression for users in 12-hour-preferring countries.
> 
> Sorry what's changed for 12h? It seems to only affect 24h users on Chrome who do not expect a 24h hourCycle
> 
I don't think so? It affects users in the `en` locale (ie, 12-hour-preferring countries - in particular, the United States) where we set `hour12: false`?